sql/codegen_xgboost.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"fmt"
+	"io"
+	"text/template"
+)
+
+type xgbTrainConfig struct {
+	NumBoostRound int  `json:"num_boost_round,omitempty"`
+	Maximize      bool `json:"maximize,omitempty"`
+}
+
+type xgbFiller struct {
+	IsTrain              bool
+	TrainingDatasetSQL   string
+	ValidationDatasetSQL string
+	TrainCfg             *xgbTrainConfig
+	Features             []*featureMeta
+	Label                *featureMeta
```

Comment:
I found our [filler for tensorflow](https://github.com/sql-machine-learning/sqlflow/blob/develop/sql/codegen.go#L62) in `codegen.go`,
https://github.com/sql-machine-learning/sqlflow/blob/9a0dc86b160c4248732124e92ea64edcfc72a41f/sql/codegen.go#L67-L69
`Features` named to `X` and `Label` to `Y`.
I would suggest such consistent naming.

Reply:
Maybe a better way is reuse the filler in `codegen`, would optimize it in the next PR.